core/client/hdfs/src/main/java/alluxio/hadoop/AlluxioHdfsInputStream.java

```diff
@@ -50,7 +51,20 @@ public int read() throws IOException {
 
   @Override
   public int read(ByteBuffer buf) throws IOException {
-    return mInput.read(buf);
+    if (mInput.getWrappedStream() instanceof ByteBufferReadable) {
+      return mInput.read(buf);
```

Comment:
Add a link to Hadoop `FSDataInputStream`

Reply:
Done